builder/proxmox/common/config.go

```diff
@@ -57,6 +57,7 @@ type Config struct {
 	VGA            vgaConfig      `mapstructure:"vga"`
 	NICs           []nicConfig    `mapstructure:"network_adapters"`
 	Disks          []diskConfig   `mapstructure:"disks"`
+	Serials        []string       `mapstructure:"serials"`
```

Comment:
Given the limitations described in the documentation, we can probably add a few checks in the `Prepare` step for the plugin, so we ensure that only valid values are accepted before we attempt to create the devices.

Like ensuring there are 4 or less elements, and that their values are the expected shape (assuming this is relevant?)

Also I wonder, why the limit of 4? Is it something enforced by the API?

Reply:
Yes, The limit of 4 is imposed by the API. 
In case of using more than 4  the build fails with `Error creating VM: 400 Parameter verification failed` message.
During the next days I will work on your suggestion during my available spare time.

Reply:
I've just pushed 2 commits and rebased the branch.
From my side I tested the changes manually the validation.

Reply:
Thanks @mabeett, I left a review at the same time, just a few nits to address, regarding testing I think if we could validate at least the `Prepare` stuff automatically that'd be best.
Maybe consider adding this to `builder/proxmox/common/config_test.go`? Other than that, we can merge this, if you validated manually that's a good start.